### PR TITLE
fix: txMiningServiceUrl lack of declaration on customNetwork

### PR DESCRIPTION
**Pull request by alexruzenhack**
### Acceptance Criteria
- It should declare `txMiningServiceUrl` on customNetwork payload to update the state on redux store
- It should config `txMiningServiceUrl` and `explorerServiceUrl` on wallet initialization, prior to the wallet start command
- it should fix network identification on network settings update
- It should filter out HTR token from fetch balance over registered tokens iteration
- It should improve saga/tokens instrumentation to make debug easier

#### Updating txMiningServiceUrl

The following demo was recorded after applying this fix to the `feat/rpc-lib-integration` branch.

https://github.com/HathorNetwork/hathor-wallet-mobile/assets/5992210/ca9f9534-c83c-4d37-80c0-46de2e9590d6

### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

**Review by pedroferreira1 — APPROVED** [review]

**Review by andreabadesso — APPROVED** [review]

File: src/sagas/networkSettings.js [hunk 1]
```diff
@@ -202,12 +202,11 @@ export function* updateNetworkSettings(action) {
...
-  } else if (potentialNetwork.startsWith(NETWORK_TESTNET)) {
+  } else if (potentialNetwork.indexOf(NETWORK_TESTNET) >= 0) {
     network = NETWORK_TESTNET;
-  } else if (potentialNetwork.startsWith(NETWORK_PRIVATENET)) {
+  } else if (potentialNetwork.indexOf(NETWORK_PRIVATENET) >= 0) {
     network = NETWORK_PRIVATENET;
   } else {
-    log.debug('The network informed is not allowed. Make sure your network is either "mainnet", "testnet" or "privatenet", or starts with "testnet" or "privatenet".');
     yield put(networkSettingsUpdateFailure());
     return;
```
Comment (pedroferreira1) [inline src/sagas/networkSettings.js:211]: I've been using the following code in the desktop (and in the lib method that will substitute this code here)

```
if (potentialNetwork === NETWORK_MAINNET) {
  network = NETWORK_MAINNET;
} else if (potentialNetwork.includes(NETWORK_TESTNET)) {
  network = NETWORK_TESTNET;
} else {
  network = NETWORK_PRIVATENET;
}
```

I think we can continue with your code but there's a small difference and the code I wrote above will substitute this one soon with the lib method.

**Review by pedroferreira1 — APPROVED** [review]

File: src/sagas/wallet.js [hunk 2]
```diff
@@ -337,7 +341,7 @@ export function* loadTokens() {
   // Since we already know here what tokens are registered, we can dispatch actions
   // to asynchronously load the balances of each one. The `put` effect will just dispatch
   // and continue, loading the tokens asynchronously
-  for (const token of registeredUids) {
+  for (const token of registeredUids.filter((token) => token !== htrUid)) {
```
Comment (andreabadesso) [inline src/sagas/wallet.js:344]:
```suggestion
  for (const token of registeredUids) {
    if (token !== htrUid) continue;
```

File: src/sagas/tokens.js [hunk 3]
```diff
@@ -96,6 +96,8 @@ function* fetchTokenBalance(action) {
     const tokenBalance = yield select((state) => get(state.tokensBalance, tokenId));
 
     if (!force && tokenBalance && tokenBalance.oldStatus === TOKEN_DOWNLOAD_STATUS.READY) {
+      log.debug(`Token download status READY.`);
+      log.debug(`Success fetching token balance for token ${tokenId}.`);
```
Comment (andreabadesso) [inline src/sagas/tokens.js:100]:
```suggestion
      log.debug(`Token balance already downloaded for ${tokenId}. Will not download again.`);
```

File: src/sagas/tokens.js [hunk 4]
```diff
@@ -131,15 +133,15 @@ function* fetchTokenHistory(action) {
 
     if (!force && tokenHistory && tokenHistory.oldStatus === TOKEN_DOWNLOAD_STATUS.READY) {
       // The data is already loaded, we should dispatch success
-      log.debug('Success fetching token history from store.');
+      log.debug(`Success fetching token history from store for token ${tokenId}.`);
```
Comment (andreabadesso) [inline src/sagas/tokens.js:136]: Same here

**Review by andreabadesso — CHANGES_REQUESTED** [review]

**Review by andreabadesso — APPROVED** [review]

**Pull request merged**